cpp/tests/utilities/column_utilities.cu

```diff
@@ -209,18 +204,19 @@ struct column_comparator_impl {
                                               corresponding_rows_unequal,
                                               corresponding_rows_not_equivalent>;
 
-    // worst case - everything is different
-    thrust::device_vector<int> differences(lhs.size());
-
-    auto diff_iter = thrust::copy_if(thrust::device,
+    auto differences = thrust::device_vector<int>(lhs.size());  // worst case: everything different
+    auto diff_iter   = thrust::copy_if(thrust::device,
                                      thrust::make_counting_iterator(0),
                                      thrust::make_counting_iterator(lhs.size()),
                                      differences.begin(),
                                      ComparatorType(*d_lhs, *d_rhs));
 
-    // shrink back down
-    differences.resize(thrust::distance(differences.begin(), diff_iter));
-    print_differences(differences, lhs, rhs, print_all_differences, depth);
+    differences.resize(thrust::distance(differences.begin(), diff_iter));  // shrink back down
+
+    if (not differences.empty()) {
+      auto const msg = differences_message(differences, lhs, rhs, print_all_differences, depth);
+      EXPECT_EQ(differences.size(), size_t{0}) << msg;
```

Comment:
This `EXPECT_EQ` will always fail inside this `if` block.

Reply:
That was the behaviour before as well. Is there a Google Test `FAIL` macro that can be used instead?

Reply:
`ASSERT_FALSE`

Reply:
Replaced with `GTEST_FAIL`